src/corva/api_adapter.py

```diff
@@ -158,6 +158,31 @@ def insert(
 
         return InsertResult.parse_obj(response.json())
 
+    def replace(self, provider: str, dataset: str, id_: str, *, document: dict) -> dict:
+        """Replaces all document data.
+
+        Replace all document data using the endpoint PUT
+            'data/{provider}/{dataset}/{id}/'.
+
+        Args:
+            provider: company name owning the dataset.
+            dataset: dataset name.
+            id_: document id to replace.
+            document: new document data.
+
+        Raises:
+            httpx.HTTPStatusError: if request was unsuccessful.
+
+        Returns:
+            Updated document.
+        """
+
+        response = self.http.put(url=f"data/{provider}/{dataset}/{id_}/", json=document)
```

Comment:
is it technically possible for one company provide company name of another company and replace their document?

Reply:
This is regulated by permissions of the company making the request. As far as I know company A can request access to `read` some dataset of company B.

Reply:
so, it should be technically possible within the sdk and "permission regulating" part is in place in data-api. Correct?